tortoise/verifying_algorithm.go

```diff
@@ -39,14 +33,32 @@ type Config struct {
 	RerunInterval   time.Duration // how often to rerun from genesis
 }
 
+// ThreadSafeVerifyingTortoise is a thread safe verifying tortoise wrapper, it just locks all actions.
+type ThreadSafeVerifyingTortoise struct {
+	// update for rerun
+	update atomic.UnsafePointer
```

Comment:
we can also protect this struct with `mu` right? it seems a simple write lock at the end of the `rerun()`
i don't see why this field should be different from others. code consistency out-weighs performance gain here.